### PR TITLE
Feat/push notification

### PR DESCRIPTION
**Pull request by sheralim012**
- Initial setup for push notifications
- user will receive a notification of a new message from the chatbot

### CODE REVIEW

File: home/processors.py [hunk 1]
```diff
@@ -13,3 +13,9 @@ def show_footers(request):
     if start_link:
         show_footers_ = False
     return {'show_footers': show_footers_}
+
+
+def add_vapid_public_key(request):
```
Comment (istride) [inline home/processors.py:18]: would it be better to keep this in the messaging module?

File: messaging/chat.py [hunk 2]
```diff
@@ -35,6 +36,14 @@ def _record_message_in_database(self, sender, rapidpro_message_id, text, quick_r
...
                 message.text = f'{message.text}{text}'
                 message.save(update_fields=['text'])
+            else:
+                payload = {
+                    'head': 'New Message!',
+                    'body': 'Click here to view it.',
+                    'url': self.thread.get_absolute_url(),
+                }
+                for user in self.thread.users.all():
+                    send_user_notification(user=user, payload=payload, ttl=1000)
```
Comment (istride) [inline messaging/chat.py:46]: Is it possible for `send_user_notification` to fail? Does it throw exceptions that we may need to handle?

**Review by istride — CHANGES_REQUESTED** [review]
I'm worried by the amount of JavaScript that probably has no automated tests covering it. Is there any way to provide more coverage, for example, by adding Selenium tests?

Given that there is now much more JS in this project than before, we need to be more disciplined in our approach. What I see is a lot of global variables, which are likely to be problematic in future.

File: home/processors.py (hunk 1 above)
Comment (sheralim012) [reply home/processors.py:18]: done

File: messaging/chat.py (hunk 2 above)
Comment (sheralim012) [reply messaging/chat.py]: yes, added try-except block and log the exception

**Review by istride — APPROVED** [review]

**Pull request merged**